packages/console/src/widget.ts

```diff
@@ -292,6 +292,8 @@ class CodeConsole extends Widget {
       return this._execute(promptCell);
     }
 
+    // Add a new line and indent it before seeing if we can execute
+    promptCell.editor.newIndentedLine();
```

Comment:
I am uncomfortable with adding a new line before sending it to the kernel. We don't know what kernel is running here, and some of them will have significant whitespace. Why not do this after we know that it is incomplete?

Reply:
Sounds good. Previously, it added the newline before sending to the kernel and I was attempting to preserve that.

Reply:
Ah, interesting. Yeah, I would consider the previous behavior to be a bug (if a difficult-to-trigger one).

Reply:
Yeah, nice catch. I agree with Ian that the previous behavior is a bug.